### PR TITLE
P81-21463 - add 'get storage' API endpoint

### PR DESCRIPTION
**Pull request by lotanaharoniP81**

### CODE REVIEW

File: proxmox/client.go [hunk 1]
```diff
@@ -256,6 +256,21 @@ func (c *Client) GetVmConfig(vmr *VmRef) (vmConfig map[string]interface{}, err e
...
+	var data map[string]interface{}
+	url := fmt.Sprintf("/nodes/%s/storage/", nodeName)
+	err = c.GetJsonRetryable(url, &data, 3)
+	if err != nil {
+		return nil, err
+	}
+	if data["data"] == nil {
+		return nil, fmt.Errorf("storage STATUS not readable")
+	}
+	storageStatus = data["data"].([]interface{})
```
Comment (bartekl) [inline proxmox/client.go]: Please validate that this is assertion is successful.

**Review by bartekl — COMMENTED** [review]
inline

File: proxmox/client.go [hunk 2]
```diff
@@ -256,6 +256,21 @@ func (c *Client) GetVmConfig(vmr *VmRef) (vmConfig map[string]interface{}, err e
 	return c.GetItemConfigMapStringInterface("/nodes/"+vmr.node+"/"+vmr.vmType+"/"+strconv.Itoa(vmr.vmId)+"/config", "vm", "CONFIG")
 }
 
+// GetStorage Get status for all datastores
+func (c *Client) GetStorage(nodeName string) (storageStatus []interface{}, err error) {
+	var data map[string]interface{}
+	url := fmt.Sprintf("/nodes/%s/storage/", nodeName)
```
Comment (ranamarP81) [inline proxmox/client.go]: please don't use nodes and storage. 
use const of them and then put them in the line.

File: proxmox/client.go [hunk 3]
```diff
@@ -256,6 +256,21 @@ func (c *Client) GetVmConfig(vmr *VmRef) (vmConfig map[string]interface{}, err e
...
+	err = c.GetJsonRetryable(url, &data, 3)
```
Comment (ranamarP81) [inline proxmox/client.go]: maybe we should do the "3" - numbers of tries as configurable? 
I mean tomorrow we can think that 5 is better or 2 is better

Comment (lotanaharoniP81) [reply proxmox/client.go]: All this library is written like that, I just added another endpoint..
But I will fix it :)

**Review by ranamarP81 — APPROVED** [review]

**Review by bartekl — APPROVED** [review]

**Pull request merged**